test/common/network/dns_impl_test.cc

```diff
@@ -253,11 +253,12 @@ class TestDnsServerQuery {
   const HostMap& hosts_A_;
   const HostMap& hosts_AAAA_;
   const CNameMap& cnames_;
+  const std::chrono::seconds& record_ttl_;
```

Comment:
I realize that you didn't write most of this code, but is there any reason to have these all be references vs. just have a pointer to the parent class?

Reply:
Does the parent class in `a pointer to the parent class` mean `TestDnsServerQuery`? These data members are in `TestDnsServerQuery` actually.

Reply:
Yes, sorry, wherever these are defined. My point is that if we are just referring to something else by reference I don't think we need to specify each parameter?

Reply:
Oh. I see! I will replace those data member with the pointer. Thanks!

Reply:
I can tell why using reference here. `TestDnsServerQuery` is used by `TestDnsServer` which also contains `hosts_A_`, `hosts_AAAA_` and other stuff. In order to avoid circular include, reference is used here rather than having a pointer to `TestDnsServer` directly.

Reply:
Can't you forward declare the type or something of that sort? If not, should all of them remain references to be consistent?

Reply:
I tried to use forward declare and the compilers complaint incomplete type.

Reply:
I will keep all of them reference.